### PR TITLE
Ozaria Encouragement modal

### PR DESCRIPTION
**Pull request by jmif**
- Adds BaseModal and BaseModalContainer from ozaria
- Adds encouragement modal experiment
- Adds encouragement modal HTML / CSS

### CODE REVIEW

File: app/views/teachers/OzariaEncouragementModal.vue [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+                    <span>{{ $t('teacher_ozaria_encouragement_modal.you_can_return') }}</span>
+                </div>
+            </div>
+        </template>
+    </base-modal>
+</template>
+
+<style scoped lang="scss">
+    .encouragement-modal {
+        /deep/ .modal-container {
```
Comment (codeluggage) [inline app/views/teachers/OzariaEncouragementModal.vue]: Same thoughts on `/deep/` as above, in this file.

File: app/views/courses/TeacherClassesView.coffee [hunk 2]
```diff
@@ -260,6 +279,14 @@ module.exports = class TeacherClassesView extends RootView
         @calculateQuestCompletion()
         @render()
 
+  openOzariaEncouragementModal: () ->
+    # TODO clean this up when modal is closed or page is unloaded
+    # The modal container needs to exist outside of $el because the loading screen swap deletes the holder element
```
[the remark this answers is not in the document]
Comment (jmif) [reply app/views/courses/TeacherClassesView.coffee]: Yep it's naturally handled by Vue and I was hoping it'd also be handled here but there is a race condition with the loading bar that ends up deleting the container early.  I actually did this and forgot to remove TODO - removing.

File: app/components/common/BaseModal.vue [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      <div class="modal-footer">
+        <slot name="footer" />
+      </div>
+    </div>
+  </base-modal-container>
+</template>
+
+<style lang="sass">
+  .modal
+    /deep/ .container
```
[the remark this answers is not in the document]
Comment (jmif) [reply app/components/common/BaseModal.vue]: Updated and great to know.

File: app/views/teachers/OzariaEncouragementModal.vue (hunk 1 above)
Comment (jmif) [reply app/views/teachers/OzariaEncouragementModal.vue]: Updated

File: app/components/common/BaseModal.vue [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      width: 100%
+      height: 100%
+
+      padding: 25px
+      border-radius: 10px
+
+      // transition: all .3s ease;
```
[the remark this answers is not in the document]
Comment (jmif) [reply app/components/common/BaseModal.vue]: Hm good question - this was copied over from Ozaria so I'm not sure.  I'll uncomment and make sure it does what I expect / check with Shasha.

File: app/views/courses/TeacherClassesView.coffee [hunk 5]
```diff
@@ -176,6 +178,16 @@ module.exports = class TeacherClassesView extends RootView
         container: dot
       })
 
+  destroy: ->
+    @cleanupEncouragementModal()
+
```
[the remark this answers is not in the document]
Comment (jmif) [reply app/views/courses/TeacherClassesView.coffee]: Yeah I usually use newlines to separate the different logical blocks of code, but this method is so short and there are no brackets so it looks like a floating super.  Removed.

File: app/components/common/BaseModal.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (jmif) [reply app/components/common/BaseModal.vue]: Doesn't seem to do anything actually, removed

**Pull request merged**